16 Code cleanup in IRVE

nickchan2 opened the pull request.

nickchan2 committed:
16 Cleanup

nickchan2 committed:
Renaming

Commit:
#16 attempt to fix broken branch

nickchan2 committed:
16 Cleanup

nickchan2 committed:
Merge branch '16-code-cleanup-in-irve' of https://github.com/angry-goose-initiative/irve into 16-code-cleanup-in-irve

nickchan2 committed:
Cleanup

nickchan2 committed:
Merge branch 'staging' into 16-code-cleanup-in-irve

nickchan2 committed:
16 cleanup

nickchan2 committed:
16 Fix merge

nickchan2 committed:
16 Cleanup

nickchan2 committed:
16 Fix?

nickchan2 committed:
16 More cleanup (but still broken from renaming csr files)

nickchan2 committed:
Rename CSR.cpp to csr.cpp

nickchan2 committed:
Rename CSR.h to csr.h

nickchan2 committed:
16 Fixed?

nickchan2 committed:
16 Cleanup

nickchan2 committed:
16 Cleanup

nickchan2 committed:
16 Cleanup

nickchan2 committed:
16 Cleanup

nickchan2 committed:
16 Cleanup

JZJisawesome commented:
This looks pretty good! As discussed we've removed the redundant namespaces, improved class names, and renamed CSR.cpp and .h.

Just a few things to consider:
- Some of the dividers for Includes, etc seemed to be accidentally removed. Like from aclint.cpp, for example,
```
/* ------------------------------------------------------------------------------------------------
 * Includes
 * --------------------------------------------------------------------------------------------- */
 ```
is missing (no big deal :)

- I don't know how I feel about removing the filename from the top of each file. I know we both agreed earlier to do so but I'm kind of regretting it as it makes it harder to grep. Let me know what you think Nick.

- `Wordo_write` should be `word_to_write` in the `execute::amo` function I think?

- If you renamed `rvexception` to `rv_trap` we should probably rename the file to match as well?

- To go a bit easier on our git history do you think it would make sense to squash some of the commits together?

On my end I do need to make sure that the kernel still works, and I'll try to do that tomorrow. Seeing that all of the unit tests pass though I don't anticipate any issues. I just want to be extra safe :)

nickchan2 commented:
> This looks pretty good! As discussed we've removed the redundant namespaces, improved class names, and renamed CSR.cpp and .h.
> 
Yeah still a bit left but done for the most part at this point
> Just a few things to consider:
> 
> * Some of the dividers for Includes, etc seemed to be accidentally removed. Like from aclint.cpp, for example,
> 
> ```
> /* ------------------------------------------------------------------------------------------------
>  * Includes
>  * --------------------------------------------------------------------------------------------- */
> ```
> 
> is missing (no big deal :)
> 
Yeah my bad, I was trying out removing them in places where they weren't helpful (mainly class headers) and forgot that they were still there. I'll add them back if you think they should be there
> * I don't know how I feel about removing the filename from the top of each file. I know we both agreed earlier to do so but I'm kind of regretting it as it makes it harder to grep. Let me know what you think Nick.
>
What does it mean to 'grep'?
> * `Wordo_write` should be `word_to_write` in the `execute::amo` function I think?
>
Oh yeah, the perils of mass find and replace, good catch
> * If you renamed `rvexception` to `rv_trap` we should probably rename the file to match as well?
>
Yeah I'm aware of that, it will be fixed in a later PR for this same ticket
> * To go a bit easier on our git history do you think it would make sense to squash some of the commits together?
> 
> On my end I do need to make sure that the kernel still works, and I'll try to do that tomorrow. Seeing that all of the unit tests pass though I don't anticipate any issues. I just want to be extra safe :)
>
Sounds good, keep me updated!

JZJisawesome commented:
Alrighty the mmode kernel still seems to work!

Grep is basically a multi-file text search (most of the time). It can be convenient sometimes when searching for a file on the command line (if the file name is in the file).

At the end of the day though having the filename in the file is more of a preference thing for me, and given you put in the effort already I think we'll just leave it :)

We can squash the commits when we merge so no worries there, and you mentioned deferring the `rvexception` renaming to later.

So given all of that I think all that is left is the `amo` function fix!
